### PR TITLE
Define domain models

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Related Issue(s)
- #163 163

## Verification
- [x] **Your** code builds clean without any errors or warnings

### CODE REVIEW

File: src/Altinn.Notifications.Core/Altinn.Notifications.Core.csproj [hunk 1]
```diff
@@ -7,8 +7,23 @@
...
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <AdditionalFiles Include="..\..\stylecop.json">
+      <Link>stylecop.json</Link>
+    </AdditionalFiles>
   </ItemGroup>
 
+  <PropertyGroup>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>$(NoWarn);1591</NoWarn>
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Altinn.Notifications.Core.csproj:22]: Probably don't need the NoWarn tag. We want XML comment on public visible types and members.
`1591: Missing XML comment for publicly visible type or member 'Type_or_Member'`

File: src/Altinn.Notifications.Core/Altinn.Notifications.Core.csproj [hunk 2]
```diff
@@ -7,8 +7,23 @@
     <ProjectGuid>{607C02FB-8C76-41BD-A3DF-282B2D0E2296}</ProjectGuid>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.1" />
+<ItemGroup Condition="'$(Configuration)'=='Debug'">
+    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435">
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Altinn.Notifications.Core.csproj:11]: There is a new version 507.

File: src/Altinn.Notifications.Core/Models/Creator.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿namespace Altinn.Notifications.Core.Models;
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Models/Creator.cs:1]: We're inconsistent with the use of file-scoped namespaces. We can add an .editorconfig file where we turn it on.

File: src/Altinn.Notifications.Core/Models/EMail.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
+﻿using Altinn.Notifications.Core.Enums;
+
+namespace Altinn.Notifications.Core.Models
+{
+    /// <summary>
+    /// Class representing an email
+    /// </summary>
+    public class EMail
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Models/EMail.cs:8]: If we consider email as a word, the casing should be Email.

File: src/Altinn.Notifications.Core/Models/EMail.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+        /// <summary>
+        /// Gets the content type of the email.
+        /// </summary>
+        public EMailContentType ContentType { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EMail"/> class.
+        /// </summary>
+        public EMail(string id, string subject, string body, string toAddress, EMailContentType contentType)
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Models/EMail.cs:38]: SHould we add from address as well?

File: src/Altinn.Notifications.Core/Models/Creator.cs (hunk 3 above)
Comment (tba76) [reply src/Altinn.Notifications.Core/Models/Creator.cs:1]: Updated .editorconfig with file_scoped:suggestion for namespace declarations

**Review by tba76 — APPROVED** [review]

**Pull request merged**